eos-installer-data/com.endlessm.Installer.desktop.in.in

```diff
@@ -7,3 +7,4 @@ Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;System;
 OnlyShowIn=GNOME;
+NoDisplay=true
```

Comment:
I don't quite understand why we have a `NoDisplay` attribute here - wouldn't we want the launcher for the installer to be visible in things like the search? Or I guess this attribute is dynamically removed on live sessions and not for installed ones, right?

Also, what happens if a user accidentally launches the installer on an installed session? Should the installer detect that it is not in a live boot and bail out accordingly? (I suppose we already had this problem when the desktop file was in /usr/share/gdm/greeter/applications, though it came to mind just now).

Reply:
> I don't quite understand why we have a NoDisplay attribute here - wouldn't we want the launcher for the installer to be visible in things like the search? Or I guess this attribute is dynamically removed on live sessions and not for installed ones, right?

Right: we only want the reformatter to be visible in live sessions. This is the purpose of `NoDisplay=true` here. There is a [corresponding patch in the g-i-s PR](https://github.com/endlessm/gnome-initial-setup/pull/227/commits/1d67e2874a2c8e56880f6ddb0903f19ec68a5e71) which removes `NoDisplay=true` on live boots. Previously, it was hidden from non-live sessions by putting into a non-standard path.

Reply:
> Also, what happens if a user accidentally launches the installer on an installed session? Should the installer detect that it is not in a live boot and bail out accordingly? (I suppose we already had this problem when the desktop file was in /usr/share/gdm/greeter/applications, though it came to mind just now).

![screenshot from 2018-02-26 09-38-15](https://user-images.githubusercontent.com/86760/36663318-1b80618e-1ad9-11e8-822b-a054562a711d.png)

Reply:
Heh, fair enough. I guess it'd be nice have a nicer error for what the problem actually is, but that's outside the scope of this PR.